.grit/patterns/hathora_to_new_ts_sdk.md

```diff
@@ -0,0 +1,226 @@
+# Upgrade Hathora to Dedicated TS SDK
+
+```
+engine marzano(0.1)
+language js
+
+pattern replace_resource_import($resource) {
+  contains import_statement(import = import_clause(name=named_imports($imports)), source= `"@hathora/hathora-cloud-sdk"`) where {
+    if ($imports <: contains $import where $import <: $resource) {
+      $sdk = `sdk`,
+      // TODO what's the best way to ask them to provide their SDK?
+      // emit a file with a TODO? what if they use relative imports?
+      // how do we know what import path to use? add that to the TODO?
+      $sdk <: ensure_import_from(source=`"./sdk"`),
```

Comment:
I don't think we need to assume the SDK is in a different file. Just create an inline instantiation of it, like we do for OpenAI [here](https://github.com/getgrit/python/blob/main/.grit/patterns/openai.md#change-openai-import-to-sync).